storage/src/mmr/mod.rs

```diff
@@ -0,0 +1,196 @@
+mod mem;
+pub use mem::{verify_proof, verify_range_proof, InMemoryMMR};
+
+use sha2::{Digest, Sha256};
```

Comment:
We should discuss whether or not we should use `cryptography::Digest` for this. I've been trying very hard to minimize the "type sprawl" across packages and would prefer to make tweaks to `Digest`/`Hasher` such that it fits here rather than reimplementing it.

https://docs.rs/commonware-cryptography/latest/commonware_cryptography/trait.Hasher.html

Reply:
If you use this interface, we can swap out the hashing logic without modifying the MMR code: https://docs.rs/commonware-cryptography/latest/commonware_cryptography/sha256/struct.Sha256.html

Reply:
(As discussed, let's see how it performs vs a `Bytes-based` implementation)

Reply:
Switched to Bytes (aka Digest) based implementation. It's 10-30% slower based on the current benchmarks, so we might want to investigate a Digest that is based on something other than Bytes in a future iteration. For now it makes sense to have this use the same abstraction though.